Mage/src/main/java/mage/abilities/effects/common/SacrificeSourceEffect.java

```diff
@@ -48,4 +51,17 @@ public boolean apply(Game game, Ability source) {
         return false;
     }
 
+
+    public SacrificeSourceEffect setSourceName(String sourceName) {
+        this.sourceName = sourceName;
+        return this;
+    }
+
+    @Override
+    public String getText(Mode mode) {
```

Comment:
I don't like overriding `getText(Mode mode)` when the mode parameter isn't needed. Instead, add back the `staticText` in the constructor and set it direclty in the `setSourceName` method as well, with no need for storing the sourceName in a field.